AEPMedia/Tests/FunctionalTests/Scenarios/AdPlayback.swift

```diff
@@ -0,0 +1,177 @@
+            MediaHit(eventType: EventType.PLAY, playhead: 1, ts: 16000),
+            MediaHit(eventType: EventType.PING, playhead: 11, ts: 26000),
+            MediaHit(eventType: EventType.SESSION_COMPLETE, playhead: 15, ts: 30000)
+        ]
+
+        //verify
+        checkHits(expectedHits: expectedHits)
+    }
+
+    func testPrerollAd_DownloadedTracker() {
+        //setup
+        createTracker(downloaded: true)
+
+        //test
+        mediaTracker.trackSessionStart(info: mediaInfoWithDefaultPreroll.toMap(), metadata: mediaMetadata)
+        mediaTracker.trackPlay()
+        mediaTracker.trackEvent(event: MediaEvent.AdBreakStart, info: adBreakInfo.toMap())
+        mediaTracker.trackEvent(event: MediaEvent.AdStart, info: adInfo.toMap(), metadata: adMetadata)
+        waitFor(time: 55000, updatePlayhead: false)
+        mediaTracker.trackEvent(event: MediaEvent.AdComplete)
+        mediaTracker.trackEvent(event: MediaEvent.AdBreakComplete)
+        // should switch to play state
+        waitFor(time: 55000, updatePlayhead: true)
+        mediaTracker.trackComplete()
+
+        let expectedSessionStartParams: [String: Any]  = [
+            Media.ID: "mediaID",
+            Media.NAME: "mediaName",
+            Media.LENGTH: 30.0,
+            Media.CONTENT_TYPE: "aod",
+            Media.STREAM_TYPE: "audio",
+            Media.RESUME: false,
+            Media.DOWNLOADED: true,
+        ]
+
+        let expectedAdBreakStartParams: [String: Any] = [
+            AdBreak.POD_FRIENDLY_NAME: "adBreakName",
+            AdBreak.POD_INDEX: 1,
+            AdBreak.POD_SECOND: 1.1
+        ]
+
+        let expectedAdParams: [String: Any] = [
+            Ad.ID: "adID",
+            Ad.NAME: "adName",
+            Ad.LENGTH: 15.0,
+            Ad.POD_POSITION: 1
+        ]
+
+        let expectedHits: [MediaHit] = [
+            MediaHit(eventType: EventType.SESSION_START, playhead: 0, ts: 0, params: expectedSessionStartParams, customMetadata: mediaMetadata),
+            MediaHit(eventType: EventType.ADBREAK_START, playhead: 0, ts: 0, params: expectedAdBreakStartParams),
+            MediaHit(eventType: EventType.AD_START, playhead: 0, ts: 0, params: expectedAdParams, customMetadata: adMetadata),
+            MediaHit(eventType: EventType.PLAY, playhead: 0, ts: 0),
+            MediaHit(eventType: EventType.PING, playhead: 0, ts: 50000),
+            MediaHit(eventType: EventType.AD_COMPLETE, playhead: 0, ts: 55000),
+            MediaHit(eventType: EventType.ADBREAK_COMPLETE, playhead: 0, ts: 55000),
+            MediaHit(eventType: EventType.PLAY, playhead: 0, ts: 55000),
+            MediaHit(eventType: EventType.PLAY, playhead: 1, ts: 56000),
+            MediaHit(eventType: EventType.PING, playhead: 51, ts: 106000),
+            MediaHit(eventType: EventType.SESSION_COMPLETE, playhead: 55, ts: 110000)
+        ]
+
+        //verify
+        checkHits(expectedHits: expectedHits)
```

Comment:
Can we add a test case for 1 sec tracking ?

Reply:
Yes I am adding more cases.